learn-data-structures-by-building-a-shopping-cart/index17.js

```diff
@@ -0,0 +1,69 @@
+const products = [
+  {
+    id: 0,
+    name: "Vanilla buttercream cupcake",
+    price: 2.99
+  },
+  {
+    id: 1,
+    name: "French Macaroon",
+    price: 3.99
+  },
+  {
+    id: 2,
+    name: "Fruit sprinkles cupcake",
+    price: 3.99
+  },
+  {
+    id: 3,
+    name: "Pink flower cupcake",
+    price: 5.99
+  }
+];
+
+class ShoppingCart {
+  constructor() {
+    this.items = [];
+    this.discountPercentage = 30;
+  }
+
+  addItem(id, products) {
+    const item = products.find(item => item.id === id);
+    this.items.push(item);
+  }
+
+  getItems() {
+    return this.items;
+  }
+  
+  getCount() {
+    return this.items.length;
+  }
+  
+  removeItem(id) {
+    const index = this.items.indexOf(id);
+    this.items.splice(index, 1);
+  }
+  
+  clearCart() {
+    this.items = [];
+  }
+  
+  applyDiscount(amount) {
+    return this.discountPercentage / 100 * amount
+  }
+}
+
+/*
+Notice that the returned amount of appliyDiscount function has a lot of numbers after the decimal.
```

Comment:
Try to break this one up into a few steps. The first can briefly explain toFixed(), the second step could have the student log an example to the console and show that the return value is now a string, and the third could teach about parseFloat().